File: application/core/src/main/java/br/com/beblue/bluerecords/core/facade/PopulaBancoUseCase.java
package br.com.beblue.bluerecords.core.facade;

import br.com.beblue.bluerecords.core.command.CadastraDiscoCommand;
import br.com.beblue.bluerecords.core.entitidade.Genero;
import br.com.beblue.bluerecords.core.usecase.DiscoUseCase;
import br.com.beblue.bluerecords.core.usecase.GeneroUseCase;

import java.util.List;

public class PopulaBancoUseCase {

    public PopulaBancoUseCase(DiscoUseCase discoUseCase, GeneroUseCase generoUseCase) {
        this.discoUseCase = discoUseCase;
        this.generoUseCase = generoUseCase;
    }

    private DiscoUseCase discoUseCase;
    private GeneroUseCase generoUseCase;

    public void populaBancoDeDados() {
        if (!discoUseCase.discosForamCadastrados()) {
            List<Genero> generos = generoUseCase.buscarGenero();
            generos.forEach(genero -> {
                List<CadastraDiscoCommand> commands = discoUseCase.consultarDiscosParaCadastroPorGenero(genero);
                commands.forEach(it -> discoUseCase.cadastrarDisco(it));
            });
        }
    }
}

File: application/core/src/main/kotlin/br/com/beblue/bluerecords/core/usecase/PopulaBancoUseCase.kt
package br.com.beblue.bluerecords.core.usecase

import br.com.beblue.bluerecords.core.usecase.disco.DiscoUseCase
import br.com.beblue.bluerecords.core.usecase.genero.GeneroUseCase

class PopulaBancoUseCase(private val discoUseCase: DiscoUseCase, private val generoUseCase: GeneroUseCase) {

    fun populaBancoDeDados() {
        if (!discoUseCase.discosForamCadastrados()) {
            val generos = generoUseCase.buscarGenero()
            generos.forEach { genero ->
                val commands = discoUseCase.consultarDiscosParaCadastroPorGenero(genero)
                commands.forEach {
                    discoUseCase.cadastrarDisco(it)
                }
            }
        }
    }

}